src/recode.jl

```diff
@@ -368,3 +370,9 @@ function recode(a::CategoricalArray{S, N, R}, default::Any, pairs::Pair...) wher
     end
     recode!(dest, a, default, pairs...)
 end
+
+function Missings.replace(a::CategoricalArray{S, N, R, V, C}, replacement::V) where {S, N, R, V, C}
+    pool = CategoricalPool(a.pool)
```

Comment:
Doesn't `copy` work? That would be more explicit. Or even `deepcopy` here I guess.

I think we should also mark the pool as unordered if the replacement wasn't a level already, just in case (and test that). Or is it done automatically when adding a new level?

Reply:
`deepcopy` works, there is no `copy` method implemented.

I wanted this to behave like `setindex!` which uses `get!(pool, replacement)`. From what I can see new levels are just appended to the end and doesn't change `isordered`.

Should `setindex!` throw an error on new levels if they are ordered?

Reply:
Ah, I guess that's an omission. Adding a new level implicitly by assignment should change the `ordered` property since we don't know how the new level should be ordered (but it shouldn't be an error I think).